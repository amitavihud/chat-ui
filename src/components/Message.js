import React from 'react'
import styled from "styled-components";

import snorlax from './assets/001-snorlax.png'

const Styled = styled.div`
  display: grid;
  padding: 5px;
  grid-template-columns: 50px auto;

  img {
    width: 50px;
  }
`
const TextContainer = styled.div`
  text-align: left;
  display: inline-block;
  align-self: end;
`
const Bubble = styled.div`
  display: inline-block;
  margin: 8px;
  padding: 5px 7px;
  background-color: ${props => props.backgroundColor};
  color: ${props => props.color /*#76778E*/};
  border-radius: 3px;
  position: relative;
  white-space: pre-line;
`
const Arrow = styled.div`
  right: 100%;
	top: 8px;
	border: solid transparent;
	content: " ";
	height: 0;
	width: 0;
	position: absolute;
	border-color: transparent;
	border-right-color: ${props => props.backgroundColor};
	border-width: 3px;
`

export default (props) => (
  <Styled>
    <img src={snorlax} />
    <TextContainer>
      <Bubble backgroundColor={'#5152FA'} color={'#F2F2FF'}>
        <Arrow backgroundColor={'#5152FA'} />
        {props.children}
      </Bubble>
    </TextContainer>
  </Styled>
)