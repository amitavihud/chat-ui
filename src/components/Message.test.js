import React from 'react'
import renderer from 'react-test-renderer';
import Message from './Message'

describe('Message component', () => {
  it('renders the tree', () => {
    const tree = renderer.create(
      <Message 
        message={{}}
        me={false} />
    )

    expect(tree).toMatchSnapshot()
  })

  describe('it is me', () => {
    it('renders the tree correctly', () => {
      const tree = renderer.create(
        <Message
          message={{}}
          me />
      )

      expect(tree).toMatchSnapshot()
    })
  })
})