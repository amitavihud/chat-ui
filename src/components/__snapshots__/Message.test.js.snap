// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component it is me renders the tree correctly 1`] = `
<div
  className="sc-bdVaJa eMAKFS"
>
  <img
    src={undefined}
  />
  <div
    className="sc-bwzfXH gheIUh"
  >
    <div
      className="sc-htpNat jQRWOW"
      color="#F2F2FF"
    >
      <div
        className="sc-bxivhb CqVgY"
      />
      <span
        className="sc-ifAKCX BnkYH"
      >
        :
      </span>
    </div>
  </div>
</div>
`;

exports[`Message component renders the tree 1`] = `
<div
  className="sc-bdVaJa eMAKFS"
>
  <img
    src={undefined}
  />
  <div
    className="sc-bwzfXH gheIUh"
  >
    <div
      className="sc-htpNat bGHIjC"
      color="#6D6E86"
    >
      <div
        className="sc-bxivhb iwyfOA"
      />
      <span
        className="sc-ifAKCX BnkYH"
      >
        :
      </span>
    </div>
  </div>
</div>
`;
